lib/acme/client.rb

```diff
@@ -105,7 +105,7 @@ def new_order(identifiers:, not_before: nil, not_after: nil)
       identifiers
     else
       Array(identifiers).map do |identifier|
-        { type: 'dns', value: identifier }
+        identifier.is_a?(String) ? { type: 'dns', value: identifier } : identifier
```

Comment:
Is this related to the post as get thing?

Reply:
This is due to a separate change in pebble which I think ended up in 2.0.0 also.  The corresponding datatype in pebble seems to have changed from a string to a complex data type and I guess the deserialization logic changed.

https://github.com/letsencrypt/pebble/blob/master/wfe/wfe.go#L1528

https://github.com/letsencrypt/pebble/commit/bc4da68d49d78a38bfea869d38faadc8c714654c

Reply:
OK, I see.

The idea for this API was to allow to pass identifier objects or string of dns.

For example,

```ruby
new_order(identifiers: 'foorbar.com')
new_order(identifiers: ['foorbar.com', 'www.foorbar.com'])
new_order(identifiers: {type: 'dns', value: 'www.foorbar.com'})
new_order(identifiers: [{type: 'dns', value: 'www.foorbar.com'}, {type: 'dns', value: 'foorbar.com'}])
new_order(identifiers: [{type: 'other-type', value: 'value'}])
```

Not that I look into it, I doubt the L104 ever did that https://github.com/unixcharles/acme-client/blob/1aea684e29ca61b7ec7d71f9517c786173af2b27/lib/acme/client.rb#L104 

It should probably have been something like this:

```ruby
identifiers = Array(identifiers)
payload['identifiers'] = if identifiers.all? { identifiers.is_a?(String) }
  identifiers.map do |value|
    { type: 'dns', value: value }
  end
else
  identifiers
end
```

Reply:
pushed a change to support this